server/app/durablejobs/jobs/FixApplicantDobDataPathJob.java

```diff
@@ -0,0 +1,58 @@
+package durablejobs.jobs;
+
+import com.google.common.base.Preconditions;
+import durablejobs.DurableJob;
+import io.ebean.DB;
+import io.ebean.Database;
+import java.time.LocalDate;
+import models.Applicant;
+import models.PersistedDurableJob;
+import services.WellKnownPaths;
+import services.applicant.ApplicantData;
+
+/**
+ * Fixes applicant data objects that have the wrong path for applicant_date_of_birth and are causing
+ * errors when TIs attempt to apply on behalf of clients. This job finds all applicant data objects
+ * with applicant_date_of_birth set to a number and changes them to nested objects of the form
+ * {applicant_date_of_birth: {date: number}}.
+ */
+public final class FixApplicantDobDataPathJob extends DurableJob {
+  private final Database database;
+
+  private final PersistedDurableJob persistedDurableJob;
+
+  public FixApplicantDobDataPathJob(PersistedDurableJob persistedDurableJob) {
+    this.persistedDurableJob = Preconditions.checkNotNull(persistedDurableJob);
+    this.database = DB.getDefault();
+  }
+
+  @Override
+  public PersistedDurableJob getPersistedDurableJob() {
+    return persistedDurableJob;
+  }
+
+  @Override
+  public void run() {
+    String sql =
+        "SELECT * FROM applicants WHERE ((object"
+            + " #>>'{}')::jsonb)::json#>'{applicant,applicant_date_of_birth}' IS NOT NULL";
```

Comment:
suggestion (if-minor): instead of `IS NOT NULL` filter on type of the value at `'{applicant,applicant_date_of_birth}'` being a number

Reply:
done! filtered by {applicant, applicant_date_of_birth} is not null AND {applicant, applicant_date_of_birth, date} is null